application/src/main/java/org/thingsboard/server/service/install/DefaultSystemDataLoaderService.java

```diff
@@ -697,6 +697,19 @@ public void createDefaultNotificationConfigs() {
 
     @Override
     public void updateDefaultNotificationConfigs() {
+        PageDataIterable<TenantId> tenants = new PageDataIterable<>(tenantService::findTenantsIds, 500);
+        ExecutorService executor = Executors.newFixedThreadPool(Math.max(Runtime.getRuntime().availableProcessors(), 4));
+        log.info("Updating default edge failure notification configs for all tenants");
+        AtomicInteger count = new AtomicInteger();
+        for (TenantId tenantId : tenants) {
+            executor.submit(() -> {
+                notificationSettingsService.updateDefaultNotificationConfigs(tenantId);
+                int n = count.incrementAndGet();
+                if (n % 500 == 0) {
+                    log.info("{} tenants processed", n);
+                }
+            });
+        }
```

Comment:
Add `executor.shutdownNow()` and await termination, to ensure all the tenants are processed.